x/mongo/driver/operation.go

```diff
@@ -781,37 +800,39 @@ func (op Operation) retryable(desc description.Server) bool {
 
 // roundTrip writes a wiremessage to the connection and then reads a wiremessage. The wm parameter
 // is reused when reading the wiremessage.
-func (op Operation) roundTrip(ctx context.Context, conn Connection, wm []byte) ([]byte, error) {
-	err := conn.WriteWireMessage(ctx, wm)
+func (op Operation) roundTrip(ctx context.Context, conn Connection, wm *[]byte) ([]byte, error) {
```

Comment:
Echoing conversation [from slack](https://mongodb.slack.com/archives/GA296CG9L/p1661451390018179?thread_ts=1661374970.690979&cid=GA296CG9L).

Consider changing `roundTrip`, `readWireMessage` and `moreToComeRoundTrip` to follow the pattern of other functions that return the input `[]byte`. A named return may help to clarify. For example:

```go
// roundTrip writes a wiremessage to the connection and then reads a wiremessage.
// The wm parameter is expected to come from the memoryPool.
// wm is reused when reading the wiremessage.
// The returned poolSlice is expected to be returned to the memoryPool.
func (op Operation) roundTrip(ctx context.Context, conn Connection, wm []byte) (result []byte, pooledSlice []byte, err error) {
	err = conn.WriteWireMessage(ctx, wm)
	if err != nil {
		return wm, nil, op.networkError(err)
	}
	result, pooledSlice, err = op.readWireMessage(ctx, conn, wm)
	return result, pooledSlice, err 
}
```

Reply:
A returned `pooledSlice` makes sense.